pkg/cli/admin/sosreport/sosreport.go

```diff
@@ -0,0 +1,687 @@
+
+		This command will launch a pod in a temporary namespace on your cluster that gathers
+		a SOSReport from the provided node and then downloads the collected report.
+
+		Experimental: This command is under active development and may change without notice.
+	`)
+
+	sosReportExample = templates.Examples(`
+		# Collect a SOSReport using the default image configuration, writing the SOSReport into ./<host-name>.sosreport
+		  oc adm sos
+	`)
+)
+
+func NewSOSReportCommand(f kcmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewSOSReportOptions(streams)
+	cmd := &cobra.Command{
+		Use:     "sos",
+		Short:   "Launch a new instance of a pod for collecting a SOSReport for a specific node.",
+		Long:    sosReportLong,
+		Example: sosReportExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Validate())
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+
+	cmd.Flags().StringVar(&o.NodeName, "node-name", o.NodeName, "Set a specific node to use - by default a random master will be used")
+	cmd.Flags().StringSliceVar(&o.Images, "image", o.Images, "Specify a must-gather plugin image to run. If not specified, OpenShift's default must-gather image will be used.")
+
+	return cmd
+}
+
+func NewSOSReportOptions(streams genericclioptions.IOStreams) *SOSReportOptions {
+	return &SOSReportOptions{
+		SourceDir: "/tmp/sos-report",
+		IOStreams: streams,
+		LogOut:    newPrefixWriter(streams.Out, "[sosreport      ] OUT"),
+		RawOut:    streams.Out,
+		Timeout:   10 * time.Minute,
+		Images:    []string{"registry.redhat.io/rhel8/support-tools:8.4-10"},
+	}
+}
+
+func (o *SOSReportOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
+	o.RESTClientGetter = f
+	var err error
+	if o.Config, err = f.ToRESTConfig(); err != nil {
+		return err
+	}
+	if o.Client, err = kubernetes.NewForConfig(o.Config); err != nil {
+		return err
+	}
+	if o.ConfigClient, err = configclient.NewForConfig(o.Config); err != nil {
+		return err
+	}
+	if o.ImageClient, err = imagev1client.NewForConfig(o.Config); err != nil {
+		return err
+	}
+	if i := cmd.ArgsLenAtDash(); i != -1 && i < len(args) {
+		o.Command = args[i:]
+	} else {
+		o.Command = args
+	}
+	if len(o.timeoutStr) > 0 {
+		if strings.ContainsAny(o.timeoutStr, "smh") {
+			o.Timeout, err = time.ParseDuration(o.timeoutStr)
+			if err != nil {
+				return fmt.Errorf(`invalid argument %q for "--timeout" flag: %v`, o.timeoutStr, err)
+			}
+		} else {
+			fmt.Fprint(o.ErrOut, "Flag --timeout's value in seconds has been deprecated, use duration like 5s, 2m, or 3h, instead\n")
+			i, err := strconv.ParseInt(o.timeoutStr, 0, 64)
+			if err != nil {
+				return fmt.Errorf(`invalid argument %q for "--timeout" flag: %v`, o.timeoutStr, err)
+			}
+			o.Timeout = time.Duration(i) * time.Second
+		}
+	}
+	if len(o.DestDir) == 0 {
+		// TODO: Add node-name to the directory-name
```

Comment:
You may not want to do that. In some situations 'host names' or 'node names' are 'sensitive' data; and 'obfuscation' becomes an issue if your naming things this way.